// hyper-spotify/dist/components/HyperSpotifyHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HyperSpotifyHeaderFactory = undefined;

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _HyperSpotifyWidget = require('../containers/HyperSpotifyWidget');

var _HyperSpotifyWidget2 = _interopRequireDefault(_HyperSpotifyWidget);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HyperSpotifyHeaderFactory = exports.HyperSpotifyHeaderFactory = function HyperSpotifyHeaderFactory(React) {
  var HyperSpotifyWidget = (0, _HyperSpotifyWidget2.default)(React); // eslint-disable-line no-unused-vars

  return (0, _radium2.default)(function () {
    return React.createElement(
      'header',
      { style: styles.headerStyle },
      React.createElement('span', { style: styles.headerOverlayStyle }),
      React.createElement(HyperSpotifyWidget, null)
    );
  });
};

var styles = {
  'headerStyle': {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    position: 'relative',
    opacity: '0.5',
    ':hover': {
      opacity: '1'
    }
  },
  'headerOverlayStyle': {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    backgroundColor: '#000',
    opacity: 0.1
  }
};
