// hyper-spotify/dist/components/HyperSpotifyFooter.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HyperSpotifyFooterFactory = undefined;

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _HyperSpotifyWidget = require('../containers/HyperSpotifyWidget');

var _HyperSpotifyWidget2 = _interopRequireDefault(_HyperSpotifyWidget);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HyperSpotifyFooterFactory = exports.HyperSpotifyFooterFactory = function HyperSpotifyFooterFactory(React) {
  var HyperSpotifyWidget = (0, _HyperSpotifyWidget2.default)(React); // eslint-disable-line no-unused-vars

  return (0, _radium2.default)(function () {
    return React.createElement(
      'footer',
      { style: styles.footerStyle },
      React.createElement('span', { style: styles.footerOverlayStyle }),
      React.createElement(HyperSpotifyWidget, null)
    );
  });
};

var styles = {
  'footerStyle': {
    position: 'absolute',
    right: 0,
    bottom: 0,
    left: 0,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    opacity: '0.5',
    ':hover': {
      opacity: '1'
    }
  },
  'footerOverlayStyle': {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    backgroundColor: '#fff',
    opacity: 0.07
  }
};
