// hyper-spotify/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

exports.reduceUI = reduceUI;
exports.mapTermsState = mapTermsState;
exports.decorateTerms = decorateTerms;

var _HyperSpotifyHeader = require('./components/HyperSpotifyHeader');

var _HyperSpotifyFooter = require('./components/HyperSpotifyFooter');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function reduceUI(state, _ref) {
  var type = _ref.type,
      config = _ref.config;

  switch (type) {
    case 'CONFIG_LOAD':
    case 'CONFIG_RELOAD':
      {
        return state.set('hyperSpotify', config.hyperSpotify);
      }
  }

  return state;
}

function mapTermsState(_ref2, map) {
  var hyperSpotify = _ref2.ui.hyperSpotify;

  return (0, _extends3.default)({}, map, { hyperSpotify: (0, _extends3.default)({ position: 'bottom' }, hyperSpotify) });
}

function decorateTerms(Terms, _ref3) {
  var React = _ref3.React;
  var Component = React.Component;


  var HyperSpotifyHeader = (0, _HyperSpotifyHeader.HyperSpotifyHeaderFactory)(React); // eslint-disable-line no-unused-vars
  var HyperSpotifyFooter = (0, _HyperSpotifyFooter.HyperSpotifyFooterFactory)(React); // eslint-disable-line no-unused-vars

  return function (_Component) {
    (0, _inherits3.default)(_class, _Component);

    function _class() {
      (0, _classCallCheck3.default)(this, _class);
      return (0, _possibleConstructorReturn3.default)(this, (_class.__proto__ || (0, _getPrototypeOf2.default)(_class)).apply(this, arguments));
    }

    (0, _createClass3.default)(_class, [{
      key: 'render',
      value: function render() {
        var _props = this.props,
            customChildrenBefore = _props.customChildrenBefore,
            customChildren = _props.customChildren,
            position = _props.hyperSpotify.position;


        var existingChildrenBefore = customChildrenBefore ? customChildrenBefore instanceof Array ? customChildrenBefore : [customChildrenBefore] : [];
        var existingChildren = customChildren ? customChildren instanceof Array ? customChildren : [customChildren] : [];

        if (position === 'top') {
          existingChildrenBefore = existingChildrenBefore.concat(React.createElement(HyperSpotifyHeader, null));
        }

        if (position === 'bottom') {
          existingChildren = existingChildren.concat(React.createElement(HyperSpotifyFooter, null));
        }

        return React.createElement(Terms, (0, _extends3.default)({}, this.props, {
          customChildrenBefore: existingChildrenBefore,
          customChildren: existingChildren
        }));
      }
    }]);
    return _class;
  }(Component);
}
